src/app/components/Tokens/TokenTransfers.tsx

```diff
@@ -234,16 +262,23 @@ export const TokenTransfers: FC<TokenTransfersProps> = ({
           ? [
               {
                 key: 'tokenType',
-                // TODO: temporary workaround until token type becomes available as part of RuntimeEvent
-                content: <DelayedTokenTransferTokenType event={transfer} />,
+                content: transfer.evm_log_details?.token_type ? (
+                  <TokenTypeTag tokenType={transfer.evm_log_details.token_type} />
+                ) : (
+                  <DelayedTokenTransferTokenType event={transfer} />
+                ),
                 align: TableCellAlign.Center,
               },
             ]
           : []),
         {
           key: 'value',
           align: TableCellAlign.Right,
-          content: <DelayedEventBalance event={transfer} tickerAsLink={differentTokens} />,
+          content: transfer.evm_log_details?.token_type ? (
+            <EventBalance event={transfer} tickerAsLink={differentTokens} />
+          ) : (
+            <DelayedEventBalance event={transfer} tickerAsLink={differentTokens} />
```

Comment:
Can delayed versions of these components ever actually obtain more information, if backend JOIN couldn't?
Or is this so that we can deploy before API is updated, and later remove it?

Reply:
Yes, the intention behind the current code is that it should work optimally if the newly added data is present, and fall back gracefully to lower performance if it's not.

As far as I can understand, this data will _always_ be there _for transfers_, so _maybe_ we can remove the delayed codepath later, but then again, it's a but vague what do we consider to be a transfer, so I am not enterely certain about that. (But I'll try to check with the Nexus devs.)